### ISSUE TITLE
WebSocket is closed unexpectedly

### ISSUE THREAD

**Issue by fabien** [dwstech/muster#38]
**Describe the bug**

When using a `remote` node over a WebSocket connection, the connection is unexpectedly closed as soon as a 2nd request/subscription is made. If I remove the 2nd request (any of the two) it works - requesting exactly the same reference twice seems to work, too.

**To Reproduce**

The setup is exactly like the example in the documentation. Below is an adaptation of that example; however, it also failed with the exact code shown in the docs.

Server-side:

```javascript
const serverApp = muster({
    greeting: variable('Hello World!'),
    alias: ref('greeting')
});

startWebsocketServer(server, serverApp);

function startWebsocketServer(server, app) {
    var wss = new WebSocket.Server({ server: server });
    console.info('WebSocket server started...');
    wss.on('connection', function(ws) {
         socketConnect(app, ws);
    });
};

```

Client-side:

```javascript
const clientApp = muster({
    local: 'Local Value',
    api: remote('ws://localhost:8000/', { useSockets: true })
});

clientApp.resolve(ref('api', 'greeting')).subscribe(function(result) {
    console.log('Greeting: %s', result);
});

clientApp.resolve(ref('api', 'alias')).subscribe(function(result) {
    console.log('Greeting: %s', result);
});
```

**Additional context**

```
WebSocket connection to 'ws://localhost:8000/' failed: WebSocket is closed before the connection is established.
  | close | @ | socket.js:174
-- | -- | -- | --
  | (anonymous) | @ | socket-middleware.js:29
  | setData | @ | message-transport-middleware.js:56
  | setData | @ | create-node-type.js:207
  | onUnsubscribe | @ | message-transport-middleware.js:54
  | args | @ | create-node-type.js:190
  | onUnsubscribeNodeAction | @ | store.js:1233
  | releaseCachedNodeAction | @ | store.js:528
  | flush | @ | store.js:844
  | setNodeState | @ | store.js:216
  | setNodeState | @ | create-store.js:41
  | Object | @ | create-node-type.js:227
  | withTransaction | @ | with-transaction.js:10
  | setState | @ | create-node-type.js:226
  | setData.disposeGlobalEventListener.node.scope.globalEvents.listen | @ | on-global-event.js:39
  | listeners.forEach | @ | emitter.js:12
  | emit | @ | emitter.js:12
  | emit | @ | emitter.js:15
  | flush | @ | store.js:892
  | subscribe | @ | store.js:180
  | subscribe | @ | create-store.js:14
  | Object | @ | muster.js:53
  | stream | @ | stream.js:506
  | Object | @ | muster.js:91
  | stream | @ | stream.js:506
  | createStream | @ | stream.js:182
  | stream | @ | stream.js:506
  | subscribe | @ | stream.js:490

```

**Comment by fabien** [comment]
When doing those requests sequentially, it confirms the closing issue too:

```javascript
clientApp.resolve(ref('api', 'greeting')).then(function(result) {
    console.log('Greeting: %s', result);
}).then(function() {
    clientApp.resolve(ref('api', 'alias')).then(function(result) {
        console.log('Greeting: %s', result);
    });
});
```

```
socket.js:158 WebSocket is already in CLOSING or CLOSED state.
  | send | @ | socket.js:158
-- | -- | -- | --
  | send | @ | socket-middleware.js:39
  | onUnsubscribe | @ | message-transport-middleware.js:60
  | args | @ | create-node-type.js:190
  | onUnsubscribeNodeAction | @ | store.js:1233
  | releaseCachedNodeAction | @ | store.js:528
  | once | @ | store.js:183
  | (anonymous) | @ | store.js:1296
  | subscriptions.forEach | @ | take-last.js:52
  | unsubscribe | @ | take-last.js:52
  | onUnsubscribe | @ | take-last.js:58
  | args | @ | create-node-type.js:190
  | onUnsubscribeNodeAction | @ | store.js:1233
  | releaseCachedNodeAction | @ | store.js:528
  | once | @ | store.js:183
```

**Comment by fabien** [comment]
I can confirm that this happens with the client in the browser (latest Google Chrome and Safari) as well as running the client on the server (Node 9.5.0). As soon as two remote references are being resolved, the connection is dropped, actually preventing anything to be effectively resolved; there's no output.

**Comment by dws-dave** [comment]
This is definitely new behaviour - we'll investigate. Thanks for raising (and the excellent quality of the report).

**Comment by fabien** [comment]
@dws-dave here's a testing repo to reproduce the issue:

https://github.com/fabien/muster-demo

Both the client in the browser AND in Node (currently fails on Node 10.15) do not behave as expected when using the WebSocket transport. HTTP works fine (doesn't allow subscriptions to update, though). The issue seems to be pretty serious, and is currently blocking me from actually using Muster.

**Comment by dws-dave** [comment]
thanks @fabien, investigating now

**Comment by dws-dave** [comment]
@fabien think we've found the culprit - currently testing a fix internally for socket drops. will check #37 next if the fix holds.

**Comment by dws-dave** [comment]
@fabien this will be fixed in the next release, which I hope will publish tomorrow

**Comment by fabien** [comment]
@dws-dave thanks! Have a great weekend!

**Comment by dws-dave** [comment]
Oh yes, weekend 🤔

might be Monday then 😳

**Comment by fabien** [comment]
Haha, Monday will be fine of course!

**Comment by dws-dave** [comment]
Closed via #56

**Comment by fabien** [comment]
Thanks for the update - now it all works as expected!
